### PR TITLE
Defined methods in server.Device

### PR DESCRIPTION
**Pull request by bohrium272**

### CODE REVIEW

File: MOCC/src/configurations/Comm-Layer/src/server.py [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+    def push_data_ready_event(self, attr_name, counter=0):
+        """
+        Push a data ready event for the given attribute name. The event is pushed to the notification daemon.
+        The method needs only the attribue name and an optional “counter” which will be passed unchanged within the event.
+        :param str attr_name: the attribute name
+        :param int counter: the user counter
+        """
+        raise NotImplementedError()
+
+    def push_event(self, **kwargs):
```
Comment (mtambos) [inline MOCC/src/configurations/Comm-Layer/src/server.py]: ref. [comment 1]

File: MOCC/src/configurations/Comm-Layer/src/server.py [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+        raise NotImplementedError()
+    
+    def push_att_conf_event(self, attr):
+        """
+        Push an attribute configuration event
+        :param Attribute attr: the attribute for which the configuration event will be sent
+        """
+        raise NotImplementedError()
+
+    def push_change_event(self, **kwargs):
```
Comment (mtambos) [inline MOCC/src/configurations/Comm-Layer/src/server.py]: ref. [comment 1]

File: MOCC/src/configurations/Comm-Layer/src/server.py [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+    def is_there_subscriber(self, att_name, event_type):
+        """
+        Check if there is subscriber(s) listening for the event.
+        :param str att_name: the attribute name
+        :param EventType event_type: the event type
+        :return bool: True if there is at least one listener or False otherwise
+        """
+        raise NotImplementedError()
+
+    def push_archive_event(self, **kwargs):
```
Comment (mtambos) [inline MOCC/src/configurations/Comm-Layer/src/server.py]: [comment 1]: why are there parameters in the docstring that are not in the signature?

Comment (bohrium272) [reply MOCC/src/configurations/Comm-Layer/src/server.py]: These methods have multiple prototypes http://pytango.readthedocs.io/en/stable/server_api/server.html#tango.server.Device.push_archive_event
So I thought this is the best way of taking in multiple arguments

Comment (mtambos) [reply MOCC/src/configurations/Comm-Layer/src/server.py]: I think it'd be better to make the parameters explicit.
Just put a default value for the optional ones, like `attr_name=None`.
You can use `None` for all optional parameters for now.
In the case of `push_archive_event`, all versions of the method take the `attr_name` parameter, so it is not optional, must be provided by the user.
All of the above is also valid for the other two methods, `push_change_event` and `push_event`.

Comment (bohrium272) [reply MOCC/src/configurations/Comm-Layer/src/server.py]: OK. Will change.

**Pull request merged**